### DISCUSSION TITLE
Where would be the best place to start contributing more code actions?

### DISCUSSION THREAD

**Discussion by joshpetit** [kkharji/xbase#217] [Q&A]
I just started iOS dev and will be doing it for the next month or so. I was wondering where would be the best place to add codeactions similar to those in xcode (embed for example).

**Comment by kkharji** [comment]
You mean contribute code actions? if so, we need to setup builtin lsp, which I haven't had the time to do yet. But will do soon. After

**Reply by joshpetit** [reply]
Dope I see. Well I've created a few code actions using treesitter (embed in component, extracting a variable to a field in a View/struct, etc.). I use null-ls to add these as code actions. I'd be willing to contribute back these actions to xbase if we're okay with using treesitter to create these actions instead of a real language server. There's no saying we can't have both treesitter actions and lsp actions too 🤷

**Reply by kkharji** [reply]
> Dope I see. Well I've created a few code actions using treesitter (embed in component, extracting a variable to a field in a View/struct, etc.). I use null-ls to add these as code actions. I'd be willing to contribute back these actions to xbase if we're okay with using treesitter to create these actions instead of a real language server. There's no saying we can't have both treesitter actions and lsp actions too 🤷

😍😍😍 indeed, please do open a PR. You can work on lua/xbase/tressitter.lua then will see where that take us.

**Reply by joshpetit** [reply]
fantastic. I don't have any conditional logic to say if a code action should be shown or not so all of them always display, but I think it'd still be valuable to have and later when people start having issues or complaining about it build it in.

**Reply by kkharji** [reply]
Yes, we can take care of such details in the PR later. Looking forward
